routes/order_products.js

```diff
@@ -0,0 +1,54 @@
+const express = require('express');
+const orderProductsRouter = express.Router();
+
+const {getOrderById, getOrderProductById, updateOrderProduct, destroyOrderProduct} = require('../db');
+
+orderProductsRouter.patch('/:orderProductId', async (req, res, next) => {
+    const {price, quantity} = req.body;
+    const {orderProductId} = req.params;
+    const updateFields = {};
+
+    if (price) {
+        updateFields.price = price;
+    }
+
+    if (quantity) {
+        updateFields.quantity = quantity;
+    }
+
+    try {
+        const product = await getOrderProductById(Number(orderProductId));
+        const order = await getOrderById(product.orderId);
+
+        if (order.userId === req.user.id) {
+            const orderProduct = await updateOrderProduct({id: orderProductId, ...updateFields});
+        
+            res.send(orderProduct);
+        } else {
+            next({message: 'You are not authorized to access this route.'})
```

Comment:
Maybe set the status code to `403` or something to indicate what the problem is?